### PR TITLE
Log enhancements

### PR DESCRIPTION
**Pull request by dejonghb**
As @domsj predicted in  #705....

### CODE REVIEW

File: ocaml/src/alba_client_download.ml [hunk 1]
```diff
@@ -112,10 +112,16 @@ let download_packed_fragment
...
   >>== function
   | None ->
+     Lwt_log.ign_debug_f
+       ">>> download_packed_fragment: object (%S, %S) chunk %i, fragment %i from osd_id:%Li -- ENDS returning FragmentMissing <<<"
+       object_name object_id chunk_id fragment_id osd_id;
      E.fail `FragmentMissing
   | Some data ->
      osd_access # get_osd_info ~osd_id >>= fun (_, state,_) ->
      Osd_state.add_read state;
+     Lwt_log.ign_debug_f
```
Comment (toolslive) [inline ocaml/src/alba_client_download.ml]: why `ign_` ? This is already in between binds.

File: ocaml/src/asd_server.ml [hunk 2]
```diff
@@ -394,7 +394,7 @@ module Net_fd = struct
        in
        let writer buffer offset length =
          let write_from_source = Lwt_ssl.write_bytes socket buffer in
-         Lwt_extra2._write_all write_from_source offset length
+         Lwt_extra2._write_all write_from_source (identifier fd_out) offset length
```
Comment (toolslive) [inline ocaml/src/asd_server.ml]: This converts the fd to an int (and on the callee side back into an fd). 
It's not necessary and drops any type safety we have.

File: ocaml/src/osd_access.ml [hunk 3]
```diff
@@ -158,8 +158,18 @@ module Osd_pool = struct
               let open Lwt_pool2 in
               match exn with
               | Asd_protocol.Protocol.Error.Exn _ -> Keep
+              | Unix.Unix_error(Unix.EFAULT, _, _) ->
+                 Lwt_log.ign_info_f
+                   ~exn
```
Comment (toolslive) [inline ocaml/src/osd_access.ml]: drop `~exn` (it doesn't really work and you put the callstack in the message anyway)

File: ocaml/src/other/posix_stubs.c [hunk 4]
```diff
@@ -62,9 +57,7 @@ void _bs_posix_fadvise(value fd, value offset, value len, value advice) {
                         break;
         }
 
-        enter_blocking_section();
```
Comment (toolslive) [inline ocaml/src/other/posix_stubs.c]: why was the sandwich removed?

File: ocaml/src/tools/hashes.ml [hunk 5]
```diff
@@ -86,7 +86,8 @@ class crc32c_hasher =
         self # update_substring s 0 (String.length s)
 
       method update_lwt_bytes lwt_bytes pos len =
-        cur <- Alba_crc32c.Crc32c.big_array ~crc:cur lwt_bytes pos len false
+        if len > 0
```
Comment (toolslive) [inline ocaml/src/tools/hashes.ml]: The test was also added on the inside of the crc32c function ?

File: ocaml/src/tools/lwt_bytes2.ml.cppo [hunk 6]
```diff
@@ -29,14 +29,48 @@ module Lwt_bytes = struct
   let pp formatter t =
     Format.pp_print_string formatter (show t)
 
-  let unsafe_destroy (t : t) = Core_kernel.Bigstring.unsafe_destroy t
+  let raw_address (t : t) =
+    let open Ctypes in
+    let x = bigarray_start array1 t in
+    raw_address_of_ptr (to_voidp x)
+
+  let unsafe_destroy ?(msg=(Printexc.get_callstack 5 |> Printexc.raw_backtrace_to_string)) (t : t) =
```
Comment (toolslive) [inline ocaml/src/tools/lwt_bytes2.ml.cppo]: performance considerations for the evaluation of msg?

File: ocaml/src/asd_server.ml (hunk 2 above)
Comment (domsj) [inline ocaml/src/asd_server.ml]: the callee side does not turn it back into an fd

File: ocaml/src/tools/lwt_bytes2.ml.cppo (hunk 6 above)
Comment (domsj) [inline ocaml/src/tools/lwt_bytes2.ml.cppo]: that's an interesting point.
@dejonghb it probably should be written as
```
let unsafe_destroy ?msg (t : t) =
  ...
  Lwt_log.ign_debug_f "..." l address
                         (match msg with
                           | Some msg -> msg
                           | None -> Printexc.get_callstack 5 |> Printexc.raw_backtrace_to_string);
  ...
```

File: ocaml/src/other/posix_stubs.c (hunk 4 above)
Comment (dejonghb) [inline ocaml/src/other/posix_stubs.c]: I don't expect the posix_fadvise to take long enough to have any benifit of releasing/reacquiring the runtime_system, but I could be wrong here...

File: ocaml/src/tools/hashes.ml (hunk 5 above)
Comment (dejonghb) [inline ocaml/src/tools/hashes.ml]: Having the test here does not lead into the stubs part... (while running maintenance tests zero length arrays were passed in quite often)

File: ocaml/src/tools/alba_wrappers_stubs.c [hunk 7]
```diff
@@ -36,9 +36,10 @@ but WITHOUT ANY WARRANTY of any kind.
    would make sense.
 */
 CAMLprim value alba_get_maxrss(value unit) {
-    CAMLparam1 (unit);
+    CAMLparam1(unit);
+    CAMLlocal1(vres);
 
-    caml_release_runtime_system();
+    //caml_release_runtime_system();
```
Comment (domsj) [inline ocaml/src/tools/alba_wrappers_stubs.c]: if getrusage is considered to be a cheap call we should probably also remove Alba_wrappers.Sys2.lwt_get_maxrss?

File: ocaml/src/asd_server.ml (hunk 2 above)
Comment (toolslive) [inline ocaml/src/asd_server.ml]: sorry, was too eager there.

File: ocaml/src/tools/alba_wrappers_stubs.c (hunk 7 above)
Comment (toolslive) [inline ocaml/src/tools/alba_wrappers_stubs.c]: I don't think getrusage is cheap. I might be wrong too ;)

**Pull request merged**